files/grid-studies-2-2.py

```diff
@@ -7,48 +7,59 @@ class GridStudies(monome.GridApp):
     def __init__(self):
         super().__init__()
 
+    width = 0
+    height = 0
+    step = [[0 for col in range(16)] for row in range(16)]
+    play_position = -1
+
+    # when grid is plugged in via USB:
     def on_grid_ready(self):
-        self.step = [[0 for col in range(self.grid.width)] for row in range(6)]
-        self.play_position = 0
+        self.width = self.grid.width
+        self.height = self.grid.height
+        self.sequencer_rows = self.height-2
+        self.connected = True
 
-        asyncio.async(self.play())
+    # when grid is physically disconnected:
+    def on_grid_disconnect(self,*args):
+        self.connected = False
 
     async def play(self):
         while True:
-            if self.play_position == self.grid.width - 1:
+            await asyncio.sleep(0.1)
+            
+            if self.play_position == self.width - 1:
                 self.play_position = 0
             else:
                 self.play_position += 1
-
-            self.draw()
-
-            await  asyncio.sleep(0.1)
+ 
+            if self.connected:
+                self.draw()
 
     def draw(self):
-        buffer = monome.GridBuffer(self.grid.width, self.grid.height)
+        buffer = monome.GridBuffer(self.width, self.height)
 
         # display steps
-        for x in range(self.grid.width):
+        for x in range(self.width):
             # highlight the play position
             if x == self.play_position:
                 highlight = 4
             else:
                 highlight = 0
 
-            for y in range(6):
+            for y in range(self.sequencer_rows):
                 buffer.led_level_set(x, y, self.step[y][x] * 11 + highlight)
-
+                
```

Comment:
don't forget to remove trailing whitespaces :)

Reply:
oh THAT's what you meant, i'm so sorry! using the IDLE editor and have clearly been spoiled ;)